replica/table.cc

```diff
@@ -1077,6 +1077,23 @@ table::sstable_list_builder::build_new_list(const sstables::sstable_set& current
     co_return make_lw_shared<sstables::sstable_set>(std::move(new_sstable_list));
 }
 
+future<>
+compaction_group::delete_sstables_atomically(std::vector<sstables::shared_sstable> sstables_to_remove) {
+    auto f = seastar::try_with_gate(_t._sstable_deletion_gate, [this, sstables_to_remove = std::move(sstables_to_remove)] () mutable {
+        return with_semaphore(_t._sstable_deletion_sem, 1, [sstables_to_remove = std::move(sstables_to_remove)] () mutable {
+            return sstables::sstable_directory::delete_atomically(std::move(sstables_to_remove));
+        });
+    });
+
+    try {
+        co_await std::move(f);
```

Comment:
sorry, but wouldn't it be simpler to just use .handle_exception, like:
```c++
    return seastar::try_with_gate(_t._sstable_deletion_gate, [this, sstables_to_remove = std::move(sstables_to_remove)] () mutable {
        return with_semaphore(_t._sstable_deletion_sem, 1, [sstables_to_remove = std::move(sstables_to_remove)] () mutable {
            return sstables::sstable_directory::delete_atomically(std::move(sstables_to_remove));
       });
    }).handle_exception([] (std::exception_ptr ex) {
       // There is nothing more we can do here.
       // Any remaining SSTables will eventually be re-compacted and re-deleted.
       tlogger.error("Compacted SSTables deletion failed: {}. Ignored.", std::move(ex));
    });
```

The hybrid continuation chain co_await'ed in a try/catch block seems odd.